ext/libclementine-tagreader/tagreader.cpp

```diff
@@ -163,12 +164,17 @@ void TagReader::GuessArtistAndTitle(pb::tagreader::SongMetadata *song) const {
   const std::string bn = song->basefilename();
   if (!artist.empty() || !title.empty()) return;
   if (bn.empty()) return;
-  int i = bn.find(" - ");
-  if (i == std::string::npos) i = bn.find("_-_");
-  if (i == std::string::npos) return;
-  artist.assign(bn.substr(0, i));
-  title.assign(bn.substr(i + 3));
-  removeExtension(title);
+
+  static std::regex rx("^(.*)[\\s_]\\-[\\s_](.*)\\.\\w*$");
```

Comment:
In the future (not for this changeset) it might be useful if the user could specify a custom list of separators.

Btw, you can push new changes or rebase without reopening the request.

Reply:
Yeah, custom separators would be nifty.